packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -542,6 +543,11 @@ function warnIfInvokedDuringConstruction(vm: VM, methodOrPropName: string) {
         return vm.shadowRoot;
     },
 
+    get elementSelf(): Node | undefined {
```

Comment:
```suggestion
    get elementSelf(): Element {
```

I don't see how this could possibly be undefined or a `Node`. The `vm.elm` type may not be strict enough, so you might have to cast it.

Reply:
We can also put a sanity check in here:

```js
if (process.env.NODE_ENV !== 'production') {
  assert.isTrue(!isUndefined(vm.elm) && vm.elm instanceof Element, `this.elementSelf should be an Element, found: ${vm.elm}`)
}
```

Reply:
nit: just check `vm.elm instanceof Element` because `undefined instanceof Element` works just fine.

Reply:
TIL! Thanks.